pkg/platform/runtime/buildscript/buildscript.go

```diff
@@ -2,18 +2,24 @@ package buildscript
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
 	"strconv"
 	"strings"
 
 	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildexpression"
 	"github.com/alecthomas/participle/v2"
 )
 
+// Script's tagged fields will be initially filled in by Participle.
+// Expr will be constructed later and is this script's buildexpression.
 type Script struct {
-	Let *Let `parser:"'let' ':' @@"`
-	In  *In  `parser:"'in' ':' @@"`
+	Let  *Let `parser:"'let' ':' @@"`
+	In   *In  `parser:"'in' ':' @@"`
+	Expr *buildexpression.BuildExpression
```

Comment:
It's not immediately clear to me why we're now embedding the expression within the script but this feels a bit off. Ideally, these two implementations would be separate as well as the code that does the translations.

Reply:
Originally I tried just that. However, I ended up writing a translator between participle-produced Go structs and build expression Go structs. That is very brittle and any changes to the build expression structs would need to be addressed in this package too -- it's one of the problems this ticket is trying to resolve.

One solution is to put participle tags into the build expressions and construct build scripts with that, but I figured build expressions should remain distinct from build scripts.

I found it's easiest to work directly with build expressions and JSON.